change "checkout" to "use" and -HEAD to -use (#3031)
brimdata/zed: docs/lake/design.md

A branch is created with the `branch` command, e.g.,
```
zed lake branch -use logs@main staging
```
This creates a new branch called "staging" in pool "logs", which points to
the same commit object as the "main" branch.  Commits to the "staging" branch
will be added to the commit history without affecting the "main" branch
and each branch can be queried independently at any time.

The `-use` flag appears in most of the lake commands and indicates
a commit-history pointer just for that one command invocation.

> Supposing the `main` branch of `logs` was already defaulted with the "use" commmand

Reviewer: "Defaulted" reads oddly here.
```suggestion
Supposing the `main` branch of `logs` was already set as the default with the "use" command
```

zed lake branch
```

### Use

The `use` command provides a means to set the current branch context
to the indicate branch, e.g.,
```
zed lake use staging
```
When you want to specify a branch in another pool, you simply prepend
the pool name to the branch:
```
zed lake use otherpool@otherbranch
```
Just like Git `checkout -b`, you may create a new branch as a side effect

> of the `use` command with via `-b`:

Reviewer: ```suggestion
of the `use` command with `-b`:
```

i.e., `<pool>@<branch>`.  Supposing there was a branch called `updates`,
data can be committed into this branch as follows:
```
zed lake load -use logs@updates sample.zng
```

> Or, as mentioned above, you can default the branch you want to load into
> via `use`:

Reviewer: "You can default the branch" reads oddly.
```suggestion
Or, as mentioned above, you can set the default branch to load into
via `use`:
```
